src/vos/vos_obj_cache.c

```diff
@@ -368,6 +444,15 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 		D_GOTO(failed, rc = -DER_TX_RESTART);
 	}
 
+	if (!create && obj == &obj_local) {
+		/** For a read, go ahead and cache the object.  For writes, it should get released
+		 *  before the next yield.
```

Comment:
Sorry, I don't see why object shouldn't be cached for a write?

Reply:
We could but the cache is slow and the xstream won't yield between hold and release on update

Reply:
sorry, not sure I follow here? won't it hurt the update performance. as I remember first update is always slower due to the object cache miss, at least that is what I saw from the latency tracking. or I miss sth?